realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -204,6 +204,14 @@ public RealmObjectSchema removeField(String fieldName) {
         if (table.getPrimaryKey() == columnIndex) {
             table.setPrimaryKey(null);
         }
+        // When PK does not exist (-2) this cannot happen. Meanwhile a removal target has a smaller
+        // index than the PK index, we can expect an rearrangement.
+        if (columnIndex < table.getPrimaryKey()) {
+            String pkField = getPrimaryKey();
+            table.removeColumn(columnIndex);
+            table.setPrimaryKey(pkField);
```

Comment:
`invalidatePrimaryKeyIndex()` is something I was really looked into it. Then I realize that this is a patch and 1) I do not want to create a public method and 2) OS is coming.